content/articles/spatial_mgrs/index.Rmd

```diff
@@ -0,0 +1,427 @@
+
+The UTM grid lines are labelled with the corresponding coordinate values, often expressed as kilometers except in the SE corner of the map where full coordinates are given in meters, as explained by @stott_utm_1977, and by @national_geospatial-intelligence_agency_universal_2014-1 in the context of the MGRS system.
+In the case of kilometer grid labels, the (anterior) digits _before_ the last two (principal) digits are printed in superscript as they generally don't change within the map.
+This approach matches general rules of printing the various rectangular grids that emerged since World War I.
+
+```{r}
+pts <- expand.grid(
+  x = seq(50e3, 950e3, 100e3), 
+  y = 3e6 + seq(50e3, 4950e3, 100e3)
+)
+grids <- tibble(crs = 32626:32639,
+                colour = rep(viridisLite::plasma(9)[c(5,6)], 7),
+                star = rep(list(st_as_stars(pts)), 14)) |> 
+  mutate(
+    grid = purrr::map2(star, crs, function(x, crs) {
+      st_crs(x) <- st_crs(crs)
+      zone_from_30 <- crs - 32630
+      zone_bb <- st_bbox(
+        c(xmin = (zone_from_30 - 1) * 6, xmax = zone_from_30 * 6, 
+          ymin = -85, ymax = 85), 
+        crs = st_crs(4326)
+      )
+      st_as_sf(x) |> 
+        st_transform(4326) |> 
+        st_intersection(st_as_sfc(zone_bb))
+    }),
+    geom_3035 = purrr::map2(
+      grid, 
+      colour,
+      ~ geom_sf(data = st_transform(.x, 3035), colour = .y, fill = NA)
+      )
+  )
+```
+
+```{r utm-grid-europe, fig.cap="UTM grid in the European area, using cells of 100 km. As UTM zones (in one hemisphere) each have their own map projection, an angle arises between grid lines in two different UTM zones."}
+ggplot(data = countries) +
+  geom_sf(fill = "grey97", colour = "grey50") + 
+  grids$geom_3035[1:14] +
+  coord_sf(datum = "EPSG:4326", xlim = c(2.5e6, 6e6), ylim = c(1.5e6, 4.5e6)) +
+  scale_x_continuous(breaks = seq(-24, 48, 6)) +
+  scale_y_continuous(breaks = 0)
+```
+
+```{r utm-grid-belgium, fig.cap="UTM grid in the area around Belgium, using cells of 100 km."}
+ggplot(data = countries) +
+  geom_sf(fill = "grey97", colour = "grey50") + 
+  grids$geom_3035[5:8] +
+  coord_sf(datum = "EPSG:4326", xlim = c(3.5e6, 4.5e6), ylim = c(2.8e6, 3.5e6)) +
+  scale_x_continuous(breaks = seq(-24, 48, 6)) +
+  scale_y_continuous(breaks = 0)
+```
+
+The UTM grid referencing system to identify _grid cells_ at a specific resolution follows the following numerical syntax, as explained on civil maps that use it [@stott_utm_1977] [^usarmy_utm]:
+
+```
+[UTM zone number] [Truncated integer X-coordinate] [Truncated integer Y-coordinate]
+```
+
+[^usarmy_utm]: This UTM grid reference notation is also followed in the 2001 version of US Army's Field Manual 3-25.26 (Map Reading and Land Navigation) for the notation of UTM coordinates [@department_of_the_army_map_2001], of course without truncation.
+
+The spaces can be omitted or kept, or replaced by a point or a slash.
```

Comment:
Can they be omitted if the precision is >= 1m? The number of digits for X and Y is not equal.

Reply:
Good point; see https://github.com/inbo/tutorials/pull/327#discussion_r1336253036.

Reply:
> spaces can be omitted in a computational environment

That is indeed because MGRS has the strict symmetric format, and I guess it is advised because strings without spaces are shorter and a little easier to parse computationally.

NGA is not entirely clear about the 'why', so I'm just guessing actually.

> To facilitate machine-to-machine communication, an MGRS string is to have no intermediate spaces or punctuation
marks and all the letters are to be capitals.